### PR TITLE
Adds an FAQ entry why the IMU is required in 3D

### PR DESCRIPTION
**Pull request by SirVer**
This would belong into a Cartographer FAQ (compared to cartographer_ros), but having 2 FAQ's seems confusing too. Maybe we should fold the documentation of cartographer_ros into cartographers.

### CODE REVIEW

File: docs/source/faq.rst [hunk 1]
```diff
@@ -34,3 +34,25 @@ measurements, for matching. Since there are two VLP-16s, 160 UDP packets is
...
 
 __ https://github.com/googlecartographer/cartographer_ros/blob/master/cartographer_ros/configuration_files/backpack_3d.lua
+
+Why is IMU data required for 3D SLAM, but not for 2D?
+-----------------------------------------------------
+
+In 2D, Cartographer supports running the correlative scan matcher that is used
+for finding loop closure constraints also for local SLAM. It is
+computationally much more expensive than not using it, but it can often
+render odometry data unnecessary. 2D has the benefit of assuming a flat world,
```
Comment (damonkohler) [inline docs/source/faq.rst]: 2D also has ...

File: docs/source/faq.rst [hunk 2]
```diff
@@ -34,3 +34,25 @@ measurements, for matching. Since there are two VLP-16s, 160 UDP packets is
...
+so there is an implicit knowledge which way is up.
+
+In 3D, an IMU is required mainly for measuring gravity. Gravity is an
+attractive quantity to measure, since it is drift free and a very strong
+signal vs accelerations of usual SLAM platforms. Gravity is needed for two
+reasons
```
Comment (damonkohler) [inline docs/source/faq.rst]: reasons:

File: docs/source/faq.rst [hunk 3]
```diff
@@ -34,3 +34,25 @@ measurements, for matching. Since there are two VLP-16s, 160 UDP packets is
...
+-----------------------------------------------------
+
+In 2D, Cartographer supports running the correlative scan matcher that is used
+for finding loop closure constraints also for local SLAM. It is
+computationally much more expensive than not using it, but it can often
+render odometry data unnecessary. 2D has the benefit of assuming a flat world,
+so there is an implicit knowledge which way is up.
+
+In 3D, an IMU is required mainly for measuring gravity. Gravity is an
+attractive quantity to measure, since it is drift free and a very strong
```
Comment (damonkohler) [inline docs/source/faq.rst]: No comma after measure.

File: docs/source/faq.rst [hunk 4]
```diff
@@ -34,3 +34,25 @@ measurements, for matching. Since there are two VLP-16s, 160 UDP packets is
...
+signal vs accelerations of usual SLAM platforms. Gravity is needed for two
+reasons
+
+1. There are no assumptions about the world in 3D, so to properly world align the
```
Comment (damonkohler) [inline docs/source/faq.rst]: in 3D. To properly ...

File: docs/source/faq.rst [hunk 5]
```diff
@@ -34,3 +34,25 @@ measurements, for matching. Since there are two VLP-16s, 160 UDP packets is
...
+resulting trajectory and map, gravity is used to define the z-direction. This
+aligns with human expectation: up is the inverse direction of gravity.
```
Comment (damonkohler) [inline docs/source/faq.rst]: Remove the last sentence? Confusing and of little value I think.

File: docs/source/faq.rst [hunk 6]
```diff
@@ -34,3 +34,25 @@ measurements, for matching. Since there are two VLP-16s, 160 UDP packets is
...
+
+2. Roll and Pitch can be derived quite well from IMU readings, once the
```
Comment (damonkohler) [inline docs/source/faq.rst]: No comma after readings.

File: docs/source/faq.rst [hunk 7]
```diff
@@ -34,3 +34,25 @@ measurements, for matching. Since there are two VLP-16s, 160 UDP packets is
...
+direction of gravity has been established. This allows to do less work in the
```
Comment (damonkohler) [inline docs/source/faq.rst]: This saves work for the scan matcher by reducing the search window in these dimensions.

**Comment by SirVer** [comment]
@damonkohler All done. Could you have another look, please?

File: docs/source/faq.rst [hunk 8]
```diff
@@ -34,3 +34,21 @@ measurements, for matching. Since there are two VLP-16s, 160 UDP packets is
 enough for roughly 2 revolutions, one per VLP-16.
 
 __ https://github.com/googlecartographer/cartographer_ros/blob/master/cartographer_ros/configuration_files/backpack_3d.lua
+
+Why is IMU data required for 3D SLAM, but not for 2D?
+-----------------------------------------------------
+
+In 2D, Cartographer supports running the correlative scan matcher that is used for finding loop closure constraints also for local SLAM.
```
Comment (damonkohler) [inline docs/source/faq.rst]: matcher, which is normally used for finding loop closure constraints, for local SLAM.

File: docs/source/faq.rst [hunk 9]
```diff
@@ -34,3 +34,21 @@ measurements, for matching. Since there are two VLP-16s, 160 UDP packets is
...
+It is computationally much more expensive than not using it, but it can often render odometry data unnecessary.
```
Comment (damonkohler) [inline docs/source/faq.rst]: It is computationally expensive but can often render the incorporation of odometry data unnecessary.

File: docs/source/faq.rst [hunk 10]
```diff
@@ -34,3 +34,21 @@ measurements, for matching. Since there are two VLP-16s, 160 UDP packets is
...
+2D also has the benefit of assuming a flat world, so there is an implicit knowledge which way is up.
```
Comment (damonkohler) [inline docs/source/faq.rst]: world, i.e. up is implicitly defined.

File: docs/source/faq.rst [hunk 11]
```diff
@@ -34,3 +34,21 @@ measurements, for matching. Since there are two VLP-16s, 160 UDP packets is
...
+
+In 3D, an IMU is required mainly for measuring gravity.
+Gravity is an attractive quantity to measure since it does not drift and is a very strong
```
Comment (damonkohler) [inline docs/source/faq.rst]: and typically comprises most of any measured accelerations.

File: docs/source/faq.rst [hunk 12]
```diff
@@ -34,3 +34,21 @@ measurements, for matching. Since there are two VLP-16s, 160 UDP packets is
...
+signal.
+Gravity is needed for two reasons:
+
+1. there are no assumptions about the world in 3D.
```
Comment (damonkohler) [inline docs/source/faq.rst]: Capitalize

**Pull request merged**